[MIG] 10.0 porting hr_skill

jlaloux opened the pull request.

feketemihai committed:
Update traceback error with expected singleton on res.company, fix comparing apples aND ORANGES IN COMPUTE METHOS.

feketemihai committed:
Update pylint.

feketemihai committed:
Update code according to comments.

jlaloux committed:
[MIG] 10.0 porting hr_skill

jlaloux committed:
[MIG] 10.0 porting hr_skill

Yopli2k committed:
[MIG] 10.0 Porting hr_employee_age (#293)

OCA-git-bot committed:
[UPD] addons table in README.md

oca-transbot committed:
OCA Transbot updated translations from Transifex

OCA-git-bot committed:
[ADD] setup.py

oca-transbot committed:
OCA Transbot updated translations from Transifex

jlaloux committed:
[MIG] 10.0 Porting hr_contract_default_trial_length

jlaloux committed:
[MIG] 10.0 Porting hr_experience

oca-transbot committed:
OCA Transbot updated translations from Transifex

espo-tony committed:
[MIG] hr_employee_firstname: Migrated to 10.0 (#276)

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[ADD] setup.py

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

leemannd committed:
[MIG] hr_employee_phone_extension: Migrated to 10.0

leemannd committed:
[MIG] hr_emergency_contact: Migrated to 10.0

leemannd committed:
improve contact tree view with new fields

leemannd committed:
[MIG] hr_contract_reference: Migrated to 10.0

leemannd committed:
Add 10.0 hr_employee_social_media

leemannd committed:
Add 10.0 hr_employee_birth_name

feketemihai commented on hr_skill/__manifest__.py:
```diff
@@ -20,7 +20,7 @@
 
```
@jlaloux Thanks for your contribution, can you please update with small headers, add yourself in the Contributors list...

jlaloux replied:
Hi @feketemihai,
What do you mean by "update with small headers"? I don't see any python filename with big headers.
Thanks

jlaloux replied:
I supposed this was about updating copyrights?

dreispt replied:
It's OK as it is: porting a module does not grant copyright on it.

leemannd committed:
[MIG] hr_employee_seniority: Migrated to 10.0

GillesTephaneMeyomesse committed:
[MIG] hr_family to v10

leemannd committed:
add base test

leemannd committed:
update computation

feketemihai commented:
@jlaloux See comments from #299 about the menus and views...

jlaloux committed:
[UPD] Move Skills menu into HR configuration and add widget m2m_tags in employee form view and add to contributors list

jlaloux committed:
[UPD] Move Skills menu into HR configuration and add widget m2m_tags in employee form view and add to contributors list

yvaucher committed:
Merge pull request #313 from leemannd/10.0-mig-hr_employee_phone_extension

[MIG] hr_employee_phone_extension: Migrated to 10.0

jlaloux committed:
[UPD] Update copyright

jlaloux committed:
[UPD] Update copyright

jlaloux committed:
[UPD] Add to contributors, add runbot section and update copyrights

leemannd committed:
Change from fields to models for social

jlaloux committed:
[UPD] Remove useless double quotes, set sequence on Experience menu and remove useless group in views

jlaloux committed:
[UPD] Updated copyrights and remove a link into readme file

OCA-git-bot committed:
[UPD] addons table in README.md

oca-transbot committed:
OCA Transbot updated translations from Transifex

OCA-git-bot committed:
[ADD] setup.py

feketemihai commented on hr_skill/views/hr_skill.xml:
```diff
@@ -46,22 +41,8 @@
 
     <menuitem name="Skills"
-              parent="menu_hr_skill_mgt"
+              parent="hr.menu_human_resources_configuration"
```
@jlaloux Hi, thanks for the update, can you add a groups="base.group_no_one" to the menu, to only be visible when you activate the developer mode...

jlaloux replied:
Done ;)

dreispt replied:
@feketemihai AFAICT that won't be needed - the menu is only visible if the parent id.

gurneyalex committed:
Merge pull request #306 from jlaloux/migration_hr_experience

[MIG] 10.0 Porting hr_experience

gurneyalex committed:
Merge pull request #314 from leemannd/10.0-mig-hr_emergency

[MIG] hr_emergency_contact: Migrated to 10.0

gurneyalex committed:
Merge pull request #305 from jlaloux/migration_hr_contract_default_trial_length

[MIG] 10.0 Porting hr_contract_default_trial_length

gurneyalex committed:
Merge pull request #288 from feketemihai/10.0

Update hr_holidays_legal_leave errors.

jcdrubay committed:
10.0 public holidays next year (#291)

* [ADD] public holidays: add a wizard to create public holidays for the following year

* [REF] odoolint improvements

* [REF] adjustments for flake8 and pylint test

* Move public holidays menu under Leaves menu. (#290)

* [ADJ] hr_public_holidays: take into account the need of several sub-menus

* [ADD] Allow the creation of public holidays from a specific template for a specific year, and display a UserError for templates including February 29th.

* [REF] removing useless comments

* [ADJ] hr_public_holidays: Improving wizard display

* [REF] Removing tab characters

andhit-r reviewed (changes requested):
@jlaloux you may want to add unittest to pass the codecov test.

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[ADD] setup.py

gurneyalex committed:
Merge pull request #316 from leemannd/10.0-mig-hr_contract_reference

[MIG] hr_contract_reference: Migrated to 10.0

yvaucher committed:
Merge pull request #318 from leemannd/add-hr_employee_birth_name

[Add] [10.0] hr_employee_birth_name

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[ADD] setup.py

Commit:
Merge pull request #317 from leemannd/add-hr_social_media

[Add] [10.0] hr_employee_social_media

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[ADD] setup.py

oca-transbot committed:
OCA Transbot updated translations from Transifex

leemannd committed:
Calculate days & tests

gurneyalex committed:
Merge pull request #319 from leemannd/10.0-mig-hr_employee_seniority

[MIG] hr_employee_seniority: Migrated to 10.0

jlaloux committed:
[IMP] Add unit test to test nam_get function on hr_skill

jlaloux committed:
[IMP] Add unit test to test nam_get function on hr_skill

jlaloux committed:
Refactor code for pep-8

jlaloux committed:
Refactor code for pep-8

jlaloux commented:
Test added

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[ADD] setup.py

moylop260 committed:
Merge pull request #321 from acsone/10.0-mig_hr_family_mgs

[MIG] hr_family to v10

OCA-git-bot committed:
[UPD] addons table in README.md

oca-transbot committed:
OCA Transbot updated translations from Transifex

jlaloux committed:
Add group to Skill menu

jlaloux committed:
Add group to Skill menu

dreispt commented on hr_skill/views/hr_skill.xml:
```diff
@@ -43,6 +43,7 @@
               id="menu_open_view_skill_form"
-              action="open_view_skill_form"/>
+              action="open_view_skill_form"
+              groups="base.group_no_one"/>
```
This menu item will only be visible if the parent Configuration menu item also is.
So IMO this change in unnecessary.

dreispt reviewed (commented):
LGTM, only a comment about the menu items.

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[UPD] addons table in README.md

leemannd committed:
Take base from PR302

leemannd committed:
Migrate PR302 to V10.0

leemannd committed:
fix group

leemannd committed:
update to new computation method & new leave validation

leemannd committed:
Avoid error when date_from is selected

leemannd committed:
Add validation constrains

leemannd committed:
update tests

leemannd committed:
test same date

oca-transbot committed:
OCA Transbot updated translations from Transifex

gurneyalex committed:
fix computation of number_of_days

the name collision between _compute_number_of_days defined in this addon
and the method for the field number_of_days computation in the base addon hr_holidays
caused the latter to not be called => the field number_of_days would be 0 for
holidays allocation, and have the wrong sign for holidays request (because it was
forced in the the reimplemented version.

This patch:
* adds tests for the expected value of number_of_days
* adds a test for the behavior of the method for holidays allocation
* renames the method _compute_number_of_days to _recompute_number_of_days

Commit:
Merge pull request #339 from gurneyalex/10.0-fix_holidays_compute_days

[10.0] [bugfix] hr_holidays_compute_days: fix computation of number_of_days

gurneyalex committed:
Merge pull request #330 from leemannd/10.0-hr_holidays_imposed_days

[10.0][ADD] hr_holidays_imposed_days

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[ADD] setup.py

oca-transbot committed:
OCA Transbot updated translations from Transifex

astirpe committed:
Fix message notification + code review

lmignon committed:
[ADD] setup.py

lmignon committed:
[ADD] setup.py

jlaloux committed:
Merge pull request #1 from acsone/10.0-hr_skill_add_setup-lmi

[ADD] setup.py

dreispt committed:
Merge pull request #340 from onesteinbv/10_fix_hr_holidays_leave_auto_approve

[FIX][10.0] hr_holidays_leave_auto_approve avoid approval message

astirpe committed:
Update tests

dreispt committed:
Merge pull request #342 from onesteinbv/10_fix_hr_holidays_leave_auto_approve

[FIX][10.0] broken 10.0 build

alexis-via committed:
Port hr_expense_sequence to v10

alexis-via committed:
Add _rec_name = 'number' on hr.expense.sheet

oca-transbot committed:
OCA Transbot updated translations from Transifex

leemannd reviewed (approved):
Thank you for the port.
LGTM
Travis error doesn't seems to be related. 
@jlaloux Could you rebase please?

jcdrubay commented on hr_skill/tests/test_hr_skill.py:
```diff
@@ -0,0 +1,20 @@
+    def test_name_get(self):
+        hr_skill_model = self.env['hr.skill']
+        soft_skill_id = hr_skill_model.create({'name': 'Soft skill'})
+        leadership_skill_id = hr_skill_model.create({
```
You could remove the suffix _id for `soft_skill_id` and `leadership_skill_id` as those 2 variables represent an instance of object.

jcdrubay reviewed (approved):
with a suggestion of variable renaming in the test.

dreispt commented:
Can you please rebase on current 10.0 branch, to check if we get a green build?

oca-transbot committed:
OCA Transbot updated translations from Transifex

elicoidal committed:
Merge pull request #347 from akretion/10-port-hr_expense_sequence

[10.0] Port hr_expense_sequence

OCA-git-bot committed:
[UPD] addons table in README.md

oca-transbot committed:
OCA Transbot updated translations from Transifex

OCA-git-bot committed:
[ADD] setup.py

jlaloux committed:
Merge branch 'migration_hr_skill' of github.com:jlaloux/hr into migration_hr_skill

dreispt commented:
I'm sorry @jlaloux , but the change set includes files not in the module you are migrating (diffs in 351 files!). If you're having a hard time with `git rebase` I suggest you to start a new clean PR.

gurneyalex reviewed (changes requested):
there was a problem in the rebase, as pointed by @dreispt

leemannd commented:
@jlaloux If you need help for the rebase just ping me. You can go back to the original state of this PR with git reflog https://git-scm.com/docs/git-reflog

leemannd commented:
I extracted the commits left in his PR to superseed https://github.com/OCA/hr/pull/378.

pedrobaeza commented:
Superseeded by #378
